src/pg-video.vue

```diff
@@ -39,11 +54,65 @@ export default {
     if (this.slotThumbnail) {
       this.showThumbnail = true
     }
+    this.setYoutubeVideoData()
   },
   methods: {
     loadVideo () {
       this.showThumbnail = false
-      // Todo: make autoplay work.
+      this.autoplay = true
+      this.renderYoutubeVideo()
+    },
+    setYoutubeVideoData () {
+      let iframe = false
+      let url = false
+      if (this.$refs.video && this.$refs.video.children[0].children[0]) {
```

Comment:
why are you not going throuh the slot vnode? it's much cleaner that way

Reply:
Also i'd move this in the before mount if possible

Reply:
Slot has a VNode in it, but not rendered. Only a string of innerHTML so I had to use a ref instead which has the complete dom tree in it. it needs to be inside mounted, because before mounting there are no dom elements available.